frame/transaction-payment/src/lib.rs

```diff
@@ -585,53 +582,68 @@ where
 		.map(|i| (fee, i))
 	}
 
-	/// Get an appropriate priority for a transaction with the given `DispatchInfo` and fee to be
-	/// paid (with `tip`).
+	/// Get an appropriate priority for a transaction with the given `DispatchInfo`, encoded length
+	/// and user-included tip.
 	///
-	/// Values returned by the function depend on the following:
-	/// 1. The average per-weight-unit tip to be payed.
-	/// 2. The dispatch class of the transaction.
+	/// The priority is based on the amount of `tip` the user is willing to pay per unit of either
+	/// `weight` or `length`, depending which one is more limitting. For `Operational` extrinsics
+	/// we add a "virtual tip" to the calculations.
 	///
-	/// For Normal (and Mandatory) dispatchables we simply take the amount of `tip` payed
-	/// for every weight unit, scaled up by `T::WeightTipRatio`.
-	/// Operational dispatchables get an additional `priority` increase, roughly matching
-	/// a tip with amount equal to the cost required to include transaction that consumes
-	/// half of the block weight.
-	fn get_priority(info: &DispatchInfoOf<T::Call>, tip: BalanceOf<T>) -> TransactionPriority {
-		let max_block = T::BlockWeights::get().max_block;
-		let bounded_weight = info.weight.max(1).min(max_block);
-		let weight_tip_ratio = T::WeightTipRatio::get();
-		let per_weight = |val| {
-			FixedU128::saturating_from_rational(val, bounded_weight)
-				.saturating_mul_int(weight_tip_ratio)
-		};
-
-		let tip_per_weight = per_weight(tip);
+	/// The formula should simply be `tip / bounded_{weight|length}`, but since we are using
+	/// integer division, we have no guarantees it's going to give results in any reasonable
+	/// range (might simply end up being zero). Hence we use a scaling factor:
+	/// `tip * (max_block_{weight|length} / bounded_{weight|length})`, since given current
+	/// state of-the-art blockchains, number of per-block transactions is expected to be in a
+	/// range reasonable enough to not saturate the `Balance` type while multiplying by the tip.
+	fn get_priority(
```

Comment:
I also have a request: I would like to be able to call this from the outside (for https://github.com/paritytech/cumulus/pull/488). Could we make it pub?
```suggestion
	pub fn get_priority(
```

Reply:
Done :)